### DISCUSSION TITLE
support for 1password integration with desktop app

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#856] [Ideas]
### What feature would you like to see?

from their docs, 1password supports [Additional Browser integration with the desktop version of their 1Password](https://support.1password.com/connect-1password-browser-app/) **if the browser is code signed by Apple**. Zen currently doesn't have this afaik.

the alternative is to just Sign In on the browser extension as a _separate_ credential endpoint for users.

### What platform are you seeing the problem on?

macOS

### Additional context

What 1Password users currently see when attempting to install the extension into Zen

![image](https://github.com/user-attachments/assets/5d3080f8-eb34-4b5d-b409-2d6580a74a0b)

![image](https://github.com/user-attachments/assets/c48f00a2-052a-4853-917a-d657ad26c8df)

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/361_

**Comment by tommerty** [comment]
Just because the issue didn't populate this, I want to add that this affects all OS's, including Windows, though a [potential workaround](https://github.com/zen-browser/desktop/issues/361#issuecomment-2308570512) was found for Linux.

While not an absolute deal breaker for me, I'd absolutely see it get fixed

**Comment by CptnFizzbin** [comment]
What's needed to get Zen Browser set up to be code signed? It feels like that would help resolve a lot of issues around compatibility and installs.

**Reply by code-IM-perfect** [reply]
I would assume money, code signing costs a lot afaik, but open source projects do find companies who would do it for them (like the starship prompt)

**Comment by msanders** [comment]
As a temporary workaround for macOS, you can sign manually if you're already a part of the Apple Developer Program:

```
codesign --force --verify --verbose --sign "Developer ID Application" "/Applications/Zen Browser.app"
```

Then add Zen Browser to 1Password under **1Password** > **Settings…** > **Browser** > **Add Browser**.

**Comment by lissy93** [comment]
I've seen workaround for Linux and Mac, has anyone found something similar for Win?

Seems a lot of issues are stemming from the lack of code signing. Zen is awesome, and I'd love to see it succeed. If financial support will help you prioritize getting certs and signing integrated, I would be more than happy to sponsor the project.

**Comment by smoores-dev** [comment]
For anyone else on Linux that wasn't having success with the workaround posted above, in addition to the steps provided there, I found [this post](https://1password.community/discussion/129659/1password-for-firefox-refuses-to-connect-to-the-linux-desktop-app) indicating that the browser binary and its support files need to have the user:group ownership `0:0`. For me, setting this looked like `sudo chown -R 0:0 ~/.zen/browser/`.

If you're having issues, you can also try to investigate the relevant logs (info found in [this thread](https://1password.community/discussion/120954/how-the-browser-integration-works)): `cat ~/.config/1Password/logs/BrowserSupport/1Password_rCURRENT.log`

**Reply by Argiziont** [reply]
Worked like a charm. None of other options helped me, and browser connection status wasn't establidhed, untill executed below command
`sudo chown -R 0:0 /home/user/.tarball-installations/zen`

**Comment by mr-cheffy** [comment]
Macos is now signed... Do I need to submit zen somewhere to 1password?

**Reply by muffn** [reply]
It works now by adding the browser manually in 1Password by visiting the settings page in the macos 1Password app: Vault Profile Picture -> Settings -> Browser -> Add Browser 🎉 

Thanks a lot for your effort @mauro-balades

**Reply by mthuc** [reply]
Is there anything to do from a user perspective on Windows 11?

**Reply by l4kr** [reply]
Wait

**Reply by l4kr** [reply]
Or go ham on this forum
https://www.1password.community/discussions/1password/support-for-zen-browser/115811

**Reply by MarvinKohnen** [reply]
This is supposed to work for Mac right? I tried this and the extension does not work. I updated Zen (via brew) and 1Password to its latest versions, but it does not seem to help. Any ideas how to debug this?

**Reply by robpneu** [reply]
@MarvinKohnen What exactly is not working? The extension in the browser or adding the browser list of "additional browsers" in the macOS 1Password app's settings? Also, have you tried installing zen straight from https://zen-browser.app/download/ instead of brew?

**Reply by MarvinKohnen** [reply]
The extension in the browser is not working. Yes, I have tried both Zen installations. 

Current versions are MacOS Sequoia 15.7.1, Zen 1.19.2b, 1Password 8.12.6. @robpneu

**Reply by robpneu** [reply]
I'm not on the same macOS as you but I do have the same Zen and 1Password versions. Once it's been added to the list of additional browsers the extension connects to the macOS app immediately.

![Screenshot 2026-03-12 at 12 32 24 PM](https://github.com/user-attachments/assets/c348871b-2f9d-45dd-9afb-21d0ea7149e5)

**Reply by MarvinKohnen** [reply]
Reinstall of 1 password solved it for me. Thanks for your help tho!! Much appreciated @robpneu

**Comment by nikolai-in** [comment]
Any workarounds for windows?

**Reply by mil1i** [reply]
Aslo would like to see this working for Windows. But which party really needs to be the one to "fix" it? (Zen or 1Password).

With macOS, the ability to specify any browser `.app` makes it easy. Why it's not part of the 1Password Windows I don't quite understand.

**Reply by Dat-Ngu** [reply]
Unfortunately, there are no solutions for Windows 11 so far.

**Reply by Dat-Ngu** [reply]
Looks like 1password themself said the possible reason which make sense for why zen broswer is running into the issue: https://1password.community/discussion/147429/support-for-zen-browser#latest

**Comment by da411d** [comment]
Hello. Still nothing?

**Reply by smoores-dev** [reply]
Zen has done everything they can do here -- this is ultimately due to security decisions made by 1Password. Posting in this thread is unlikely to have any effect, as there's nothing more for Zen to do.

If you're on Linux or macOS, you can use these docs to configure 1Password to trust Zen Browser: https://docs.zen-browser.app/guides/1password

If you're on Windows, this is not possible: https://www.1password.community/discussions/1password/zen-browser--windows/30272/replies/30273. The only avenue for progress on this would be to ask 1Password to add support for Zen!

**Reply by da411d** [reply]
So as I understand, Zen is now properly signed, and the only limitation is 1Password to add this signature to whitelist, right?

**Reply by l4kr** [reply]
> So as I understand, Zen is now properly signed, and the only limitation is 1Password to add this signature to whitelist, right?

That's absolutely correct. The Zen developer did everything in his power to support 1password already.

https://www.1password.community/discussions/1password/support-for-zen-browser/115811

**Comment by sghng** [comment]
Follow up on this after half a year: there's an option in 1Password desktop to add integration with additional browser, and I can add Zen there by selection the app file. Does this solve the problem?

<img width="892" height="792" alt="image" src="https://github.com/user-attachments/assets/f8f8e4ff-6439-4d09-9509-1592d915d8fa" />

**Reply by sghng** [reply]
Before this, my 1P extension always logs out after quitting Zen, and now the session persists, so I guess this is indeed a solution.

**Reply by l4kr** [reply]
This has been a feature for quite a few months already and is pretty much the solution.
There's now the question of Windows integration which is 100% on 1Passwords side. But if you're not a Windows user you don't need to concern yourself anymore.

For  other Windows users who happen to stumble upon this thread, here's the 1P ticket to follow:
https://www.1password.community/discussions/1password/support-for-zen-browser/115811

It's marked as solved which is true for MacOS but **not for Windows**.

**Reply by robpneu** [reply]
It looks like it's made it into the 1Password production release on Windows 🥳 
<img width="766" height="673" alt="image" src="https://github.com/user-attachments/assets/c88ae9b4-0362-4582-8878-ae24e8d75478" />

**Reply by PikachuEXE** [reply]
Ya but it refuses files outside "Program Files"
Update 1: Won't even accept "D:\Program Files"...
Update 2: Yes "C:\Program Files" only...

**Reply by Adaliciously** [reply]
I can't get it to work myself. I added the zen.exe in Program Files, but the extension still doesnt connect to desktop app.

**Reply by PikachuEXE** [reply]
Connect, restart both, enable in zen 1P extension
All done?

**Reply by Adaliciously** [reply]
> Connect, restart both, enable in zen 1P extension All done?

doesn't seem like it
<img width="1532" height="686" alt="image" src="https://github.com/user-attachments/assets/3945e9dd-6bf1-4b29-90f1-994820155db5" />

**Reply by PikachuEXE** [reply]
Sorry dunno how to help you

**Reply by plskmtys** [reply]
Hey there!
If you have installed Zen outside of C:\Program Files, then simply copying the .exe will not work, as it will technically be a different executable than the one you are using, just with the same name. Unfortunately you have to ***install*** Zen into C:\Program Files for the integration to work. If you want to reinstall, you can turn on sync beforehand to keep most of your data.
